### ISSUE TITLE
InvalidOperationException in Lazy<T>: ValueFactory attempted to access the Value property of this instance.

### ISSUE THREAD

**Issue by JesseTG** [hadashiA/VContainer#276]
In some circumstances when `LifetimeScope.Build()` is called, one of VContainer's private `Lazy<object>`s will throw an `InvalidOperationException` upon accessing its `Value` property [as promised in its documentation](https://docs.microsoft.com/en-us/dotnet/api/system.lazy-1.value?view=netstandard-2.0#System_Lazy_1_Value). Here's my stack trace:

```
[19:38:18:120] InvalidOperationException: ValueFactory attempted to access the Value property of this instance.
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:77)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/Registration.cs:38)
VContainer.ScopedContainer+<>c__DisplayClass10_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:50)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:77)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.Resolve[T] (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:8)
CorundumGames.Chromavaders.Tests.Attributes.SimulatedInputAttribute.SimulatedInputServiceFactory (VContainer.IObjectResolver container) (at Assets/Scripts/Tests/Common/SimulatedInput/SimulatedInputAttribute.cs:87)
VContainer.Internal.FuncRegistration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/FuncRegistration.cs:34)
VContainer.ScopedContainer+<>c__DisplayClass10_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:50)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:77)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/Registration.cs:38)
VContainer.ScopedContainer+<>c__DisplayClass10_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:50)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:74)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/Registration.cs:38)
VContainer.ScopedContainer+<>c__DisplayClass10_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:50)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:74)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/Registration.cs:38)
VContainer.ScopedContainer+<>c__DisplayClass10_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:50)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.ScopedContainer.CreateTrackedInstance (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:110)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:77)
VContainer.Internal.CollectionRegistration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Internal/Registration.cs:97)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:65)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.Resolve[T] (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/IObjectResolverExtensions.cs:8)
VContainer.Unity.EntryPointDispatcher.Dispatch () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/EntryPointDispatcher.cs:34)
VContainer.Unity.EntryPointsBuilder+<>c.<EnsureDispatcherRegistered>b__0_0 (VContainer.IObjectResolver container) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/ContainerBuilderUnityExtensions.cs:20)
VContainer.ContainerBuilder.EmitCallbacks (VContainer.IObjectResolver container) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/ContainerBuilder.cs:124)
VContainer.ScopedContainerBuilder.BuildScope () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/ContainerBuilder.cs:34)
VContainer.ScopedContainer.CreateScope (System.Action`1[T] installation) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:88)
VContainer.Container.CreateScope (System.Action`1[T] installation) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Container.cs:183)
VContainer.Unity.LifetimeScope.Build () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/LifetimeScope.cs:163)
VContainer.Unity.LifetimeScope.Awake () (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/LifetimeScope.cs:130)
UnityEngine.GameObject:SetActive(Boolean)
VContainer.Unity.LifetimeScope:CreateChild(IInstaller) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/LifetimeScope.cs:193)
VContainer.Unity.LifetimeScope:CreateChild(Action`1) (at Library/PackageCache/jp.hadashikick.vcontainer@1.8.4/Runtime/Unity/LifetimeScope.cs:198)
CorundumGames.Chromavaders.Tests.Common.<<SetUpTestScope>b__7_0>d:MoveNext() (at Assets/Scripts/Tests/Common/ScopedTestFixture.cs:94)
Cysharp.Threading.Tasks.CompilerServices.AsyncUniTask`1:Run() (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/CompilerServices/StateMachineRunner.cs:189)
Cysharp.Threading.Tasks.AwaiterActions:Continuation(Object) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/UniTask.cs:21)
Cysharp.Threading.Tasks.UniTaskCompletionSourceCore`1:TrySetResult(ITestScene) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/UniTaskCompletionSource.cs:139)
Cysharp.Threading.Tasks.CompilerServices.AsyncUniTask`2:SetResult(ITestScene) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/CompilerServices/StateMachineRunner.cs:328)
Cysharp.Threading.Tasks.CompilerServices.AsyncUniTaskMethodBuilder`1:SetResult(ITestScene) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/CompilerServices/AsyncUniTaskMethodBuilder.cs:201)
CorundumGames.Chromavaders.Tests.Attributes.<LoadSceneAsyncPrivate>d__5:MoveNext() (at Assets/Scripts/Tests/Common/Attributes/Scene/SceneAddressAttribute.cs:48)
Cysharp.Threading.Tasks.CompilerServices.AsyncUniTask`2:Run() (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/CompilerServices/StateMachineRunner.cs:313)
Cysharp.Threading.Tasks.AwaiterActions:Continuation(Object) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/UniTask.cs:21)
Cysharp.Threading.Tasks.UniTaskCompletionSourceCore`1:TrySetResult(SceneInstance) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/UniTaskCompletionSource.cs:139)
Cysharp.Threading.Tasks.AsyncOperationHandleConfiguredSource`1:Continuation(AsyncOperationHandle`1) (at Library/PackageCache/com.cysharp.unitask@2.2.5/Runtime/External/Addressables/AddressablesAsyncExtensions.cs:333)
DelegateList`1:Invoke(AsyncOperationHandle`1) (at Library/PackageCache/com.unity.addressables@1.18.9/Runtime/ResourceManager/Util/DelegateList.cs:69)
UnityEngine.ResourceManagement.AsyncOperations.AsyncOperationBase`1:UpdateCallback(Single)
DelegateList`1:Invoke(Single) (at Library/PackageCache/com.unity.addressables@1.18.9/Runtime/ResourceManager/Util/DelegateList.cs:69)
UnityEngine.ResourceManagement.ResourceManager:Update(Single)
MonoBehaviourCallbackHooks:Update() (at Library/PackageCache/com.unity.addressables@1.18.9/Runtime/ResourceManager/Util/MonoBehaviourCallbackHooks.cs:26)
```

This occurs reliably, but I don't know the conditions under which it does so (or else I'd provide a minimal code sample). What information can I give you about my dependency registrations or their implementations that would be useful?

**Comment by hadashiA** [comment]
My guess is that the problem is a circular reference.

Check for references like the following：
- Resolve Class A (Parent) -> Create Child Scope.. ->  Resolve Class A (From Child)

Normally, circular references in Resolve will throw a `VContainerException`.
In this case, it is not applicable because the child scope is created during Resolve.

I did not expect to create child scopes during Resolve.
I would rather recommend that the class responsible for scope generation do it explicitly.

**Comment by JesseTG** [comment]
What would that look like?

**Comment by hadashiA** [comment]
Hmm, for example, something like this?

```csharp
class A
{
    public A(LifetimeScope currentScope)
    {
         currentScope.CreateChild(childBuilder => 
         {
             childBuilder.Register<B>(Lifetime.Scoped);
         });
    }
}

class B
{
    // To instantiate B, A is required
    // To instantiate A, Child Scope is required
    // To create a Child Scope, B is required
    // ...
    public B(A a)
    {
        /* ... */
    }
}
```

But my advice may be off the mark ?
If you have a minimal sample or something, I can look into it a bit more.

**Comment by JesseTG** [comment]
Actually, no, I got it. It turned out to be a simple circular dependency.

I resolved it by doing the following:

1. Manually walking the stack trace to find a cycle
2. Find an appropriate place to break the cycle (usually at or near the top of the stack)
3. In the constructor that causes the cycle, inject an `IObjectResolver` instead of the dependency I really want
4. Later (but *not* in the constructor itself), call `IObjectResolver.Resolve` to retrieve the dependency I really want

**Comment by hadashiA** [comment]
Im curious as to why VContainerException (circular reference error) is not detected.

It seems that VContainer is able to detect circular dependency in these cases.
https://github.com/hadashiA/VContainer/blob/master/VContainer/Assets/VContainer/Tests/Fixtures.cs#L199-L219

Please let me know if you find a way to reproduce this.

**Comment by shusei918** [comment]
I had this exception too.
The cause was also circular reference, and I already solved it.
Here is an example code to reproduce it.

```C#
using VContainer;
using VContainer.Unity;

public class GameLifetimeScope : LifetimeScope
{
    protected override void Configure(IContainerBuilder builder)
    {
        builder.Register<ModelA>(Lifetime.Singleton);
        builder.RegisterEntryPoint<ModelB>().As<IModelBInterface>();
    }
}

interface IModelBInterface
{
}

class ModelA
{
    [Inject]
    private ModelA(IModelBInterface modelBInterface)
    {
    }
}

class ModelB : IInitializable, IModelBInterface
{
    [Inject]
    private ModelB(ModelA modelA)
    {
    }

    public void Initialize()
    {
    }
}
```

And here's the stack trace.
```
InvalidOperationException: ValueFactory attempted to access the Value property of this instance.
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.Container.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:168)
VContainer.Container.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:154)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Internal/Registration.cs:38)
VContainer.Container+<>c__DisplayClass5_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:146)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.Container.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:168)
VContainer.Container.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:154)
VContainer.IObjectResolverExtensions.ResolveOrParameter (VContainer.IObjectResolver resolver, System.Type parameterType, System.String parameterName, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/IObjectResolverExtensions.cs:32)
VContainer.Internal.ReflectionInjector.CreateInstance (VContainer.IObjectResolver resolver, System.Collections.Generic.IReadOnlyList`1[T] parameters) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Internal/ReflectionInjector.cs:39)
VContainer.Internal.Registration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Internal/Registration.cs:38)
VContainer.Container+<>c__DisplayClass5_0.<.ctor>b__1 () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:146)
System.Lazy`1[T].CreateValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].LazyInitValue () (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Lazy`1[T].get_Value () (at <695d1cc93cca45069c528c15c9fdd749>:0)
VContainer.Container.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:168)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:69)
VContainer.Internal.CollectionRegistration.SpawnInstance (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Internal/Registration.cs:97)
VContainer.ScopedContainer.Resolve (VContainer.IRegistration registration) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:65)
VContainer.ScopedContainer.Resolve (System.Type type) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Container.cs:57)
VContainer.IObjectResolverExtensions.Resolve[T] (VContainer.IObjectResolver resolver) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/IObjectResolverExtensions.cs:8)
VContainer.Unity.EntryPointDispatcher.Dispatch () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Unity/EntryPointDispatcher.cs:34)
VContainer.Unity.EntryPointsBuilder+<>c.<EnsureDispatcherRegistered>b__0_0 (VContainer.IObjectResolver container) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Unity/ContainerBuilderUnityExtensions.cs:20)
VContainer.ContainerBuilder.EmitCallbacks (VContainer.IObjectResolver container) (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/ContainerBuilder.cs:124)
VContainer.ContainerBuilder.Build () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/ContainerBuilder.cs:79)
VContainer.Unity.LifetimeScope.Build () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Unity/LifetimeScope.cs:173)
VContainer.Unity.LifetimeScope.Awake () (at Library/PackageCache/jp.hadashikick.vcontainer@55d2d64a88/Runtime/Unity/LifetimeScope.cs:130)
```

**Comment by hadashiA** [comment]
Fixed #311
